.phpstorm.meta.php

```diff
@@ -49,8 +49,6 @@
     \Rector\NodeTypeResolver\Node\AttributeKey::SCOPE,
     \Rector\NodeTypeResolver\Node\AttributeKey::REPRINT_RAW_VALUE,
     \Rector\NodeTypeResolver\Node\AttributeKey::PARENT_NODE,
-    \Rector\NodeTypeResolver\Node\AttributeKey::NEXT_NODE,
-    \Rector\NodeTypeResolver\Node\AttributeKey::PREVIOUS_NODE,
```

Comment:
@samsonasik Hi Abdul, I'm trying to remove all the next/previous node and replace with direct StmtsAwareInterface.  That will help with improving memory load of the node tree as in PHPStan.

Could you help out? Take any rule and let's resolve them one by one :)

Reply:
Using `StmtsAwareInterface` doesn't make memory lower, if we still need re-connect after refactor, see:

https://github.com/rectorphp/rector-src/blob/11b278ca27f23ba4a2c55df877664fc23413a81d/src/Rector/AbstractRector.php#L441

If we need to test it out, we can remove the above call, and run tests:

```bash
vendor/bin/phpunit tests/Issues
```

and many errors will happen.

There is a condition which next/prev is needed, eg: on next of `Expr`, eg:

https://github.com/rectorphp/rector-src/blob/11b278ca27f23ba4a2c55df877664fc23413a81d/src/PhpParser/Parser/InlineCodeParser.php#L150

Reduce is possible, but removing is possibly huge BC break on end user.

Reply:
At the moment yes, we need traversing. The goal is to remove dependency on any connected nodes whatsoever. 

It might take time, but it's worth it as we will release lot of memory connections.